src/lib/query/vlan.rs

```diff
@@ -10,23 +10,21 @@ use crate::{Iface, IfaceType};
 #[derive(
     Serialize, Deserialize, Debug, PartialEq, Eq, Clone, Copy, Default,
 )]
-#[serde(rename_all = "snake_case")]
-#[non_exhaustive]
+#[serde(rename_all = "snake_case", deny_unknown_fields)]
 pub enum VlanProtocol {
     #[serde(rename = "802.1q")]
     #[default]
     Ieee8021Q,
     #[serde(rename = "802.1ad")]
-    Ieee8021AD,
-    Unknown,
+    Ieee8021Ad,
 }
 
 impl From<link::VlanProtocol> for VlanProtocol {
     fn from(d: link::VlanProtocol) -> Self {
         match d {
             link::VlanProtocol::Ieee8021Q => Self::Ieee8021Q,
-            link::VlanProtocol::Ieee8021Ad => Self::Ieee8021AD,
-            _ => Self::Unknown,
+            link::VlanProtocol::Ieee8021Ad => Self::Ieee8021Ad,
+            _ => todo!(),
```

Comment:
`todo!` is not allowed in production code.

Reply:
see above